Code/Tools/ProjectManager/Source/AddRemoteProjectDialog.cpp

```diff
@@ -0,0 +1,204 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AddRemoteProjectDialog.h>
+#include <FormFolderBrowseEditWidget.h>
+#include <TextOverflowWidget.h>
+#include <ToggleCheckbox.h>
+#include <ProjectUtils.h>
+
+#include <QVBoxLayout>
+#include <QGridLayout>
+#include <QLabel>
+#include <QLineEdit>
+#include <QDialogButtonBox>
+#include <QPushButton>
+#include <QDir>
+
+namespace O3DE::ProjectManager
+{
+    AddRemoteProjectDialog::AddRemoteProjectDialog(QWidget* parent)
+        : QDialog(parent)
+    {
+        setWindowTitle(tr("Add a remote project"));
+        setModal(true);
+        setObjectName("addRemoteProjectDialog");
+        setFixedSize(QSize(760, 600));
+
+        QVBoxLayout* vLayout = new QVBoxLayout();
+        vLayout->setContentsMargins(30, 30, 25, 10);
+        vLayout->setSpacing(0);
+        vLayout->setAlignment(Qt::AlignTop);
+        setLayout(vLayout);
+
+        QLabel* instructionTitleLabel = new QLabel(tr("Please enter a remote URL for your project"), this);
+        instructionTitleLabel->setObjectName("remoteProjectDialogInstructionTitleLabel");
+        instructionTitleLabel->setAlignment(Qt::AlignLeft);
+        vLayout->addWidget(instructionTitleLabel);
+
+        vLayout->addSpacing(10);
+
+        m_repoPath = new FormLineEditWidget(tr("Remote URL"), "", this);
+        m_repoPath->setMinimumSize(QSize(600, 0));
+        vLayout->addWidget(m_repoPath);
+
+        vLayout->addSpacing(10);
+
+        QLabel* warningLabel = new QLabel(tr("Online repositories may contain files that could potentially harm your computer,"
+            " please ensure you understand the risks before downloading from third-party sources."), this);
+        warningLabel->setObjectName("remoteProjectDialogWarningLabel");
+        warningLabel->setWordWrap(true);
+        warningLabel->setAlignment(Qt::AlignLeft);
+        vLayout->addWidget(warningLabel);
+
+        vLayout->addSpacing(10);
+
+        QFrame* hLine = new QFrame();
+        hLine->setFrameShape(QFrame::HLine);
+        hLine->setObjectName("horizontalSeparatingLine");
+        vLayout->addWidget(hLine);
+
+        vLayout->addSpacing(10);
+
+        m_downloadProjectLabel = new QLabel(tr("Download Project..."), this);
+        m_downloadProjectLabel->setObjectName("remoteProjectDialogDownloadProjectLabel");
+        m_downloadProjectLabel->setAlignment(Qt::AlignLeft);
+        vLayout->addWidget(m_downloadProjectLabel);
+
+        m_installPath = new FormFolderBrowseEditWidget(tr("Local project directory"));
+        m_installPath->setMinimumSize(QSize(600, 0));
+        vLayout->addWidget(m_installPath);
+
+        vLayout->addSpacing(10);
+
+        QHBoxLayout* buildHLayout = new QHBoxLayout(this);
+        buildHLayout->setContentsMargins(0, 0, 0, 0);
+        buildHLayout->setAlignment(Qt::AlignLeft);
+
+        m_autoBuild = new ToggleCheckbox(this);
```

Comment:
I think we can use the default QCheckbox if we use AzQtComponents `Checkbox::applyToggleSwitchStyle()`